scripts/backup-restore-test.sh

```
@@ -371,32 +291,97 @@ verify_fk_integrity() {
-                child_table,
-                parent_table
-            FROM fk_candidates
-        )
-        SELECT count(*) FROM sampled;
-    " 2>/dev/null || echo "0")
+    # Check for orphaned rows using PL/pgSQL dynamic query
+    # Iterates over each FK, builds the NOT EXISTS join from pg_attribute,
+    # and counts violations. Only runs on validated constraints to avoid
+    # false positives on intentionally unvalidated FKs.
+    log_info "Checking for orphaned rows across FK constraints..."
+    local total_orphans=0
```

Shellcheck (0.11.0)

[warning] total_orphans appears unused. Verify use (or export if used externally).

(SC2034)